devito/ir/equations/algorithms.py

```diff
@@ -160,6 +160,10 @@ def lower_exprs(expressions, **kwargs):
         mapper.update(dimension_map)
         # Add the user-supplied substitutions
         mapper.update(subs)
+        # Apply mapper to itself for nested indexing
+        mapper = {k: uxreplace(v, mapper) if k.is_Function else v
```

Comment:
does this work for doubly (or arbitrarily) nested Functions?

should this not become (somewhat) a recursive call?

going through the whole `mapper` here feels weird

Reply:
>should this not become (somewhat) a recursive call?

Yeah I think so this will only catch single recursion